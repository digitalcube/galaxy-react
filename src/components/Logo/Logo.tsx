import React, { FC } from 'react';
import { useMemo } from 'react';
import { ImageAmimotoLogo } from '..';
import { ImageDigitalcubeLogo } from '../Images/Logos/Digitalcube';
import { ImageShifterLogo } from '../Images/Logos/Shifter';

export type BrandName = 'shifter' | 'digitalcube' | 'amimoto';
export type LogoProps = {
  name: BrandName;
  width: number;
};
export const Logo: FC<LogoProps> = ({ name, width = 24 }) => {
  const aspectRatio = useMemo(() => {
    let _tmp: never;
    switch (name) {
      case 'digitalcube':
        return 15 / 77;
      case 'shifter':
        return 32 / 24;
      case 'amimoto':
        return 32 / 32;
      default:
        _tmp = name;
        return _tmp;
    }
  }, [name]);
  const height = width * aspectRatio;
  if (name === 'digitalcube') {
    return <ImageDigitalcubeLogo width={width} height={height} />;
  }
  if (name === 'amimoto') {
    return <ImageAmimotoLogo width={width} height={height} />;
  }
  return <ImageShifterLogo width={width} height={height} />;
};
