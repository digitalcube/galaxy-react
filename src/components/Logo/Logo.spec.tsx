import React from 'react';
import { render } from '@testing-library/react';
import { Default as Logo } from './Logo.stories';
import { BrandName } from './Logo';

describe('Logo', () => {
  it.each([['shifter' as const], ['digitalcube' as const]])(
    'to match snapshot [%p]',
    (name: BrandName) => {
      const { asFragment } = render(
        <Logo
          {...{
            name,
            width: 50,
          }}
        />
      );
      expect(asFragment()).toMatchSnapshot();
    }
  );
});
