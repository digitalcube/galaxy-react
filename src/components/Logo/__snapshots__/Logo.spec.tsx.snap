// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo to match snapshot ["digitalcube"] 1`] = `
<DocumentFragment>
  <img
    alt="Digitalcube"
    height="9.740259740259742"
    width="50"
  />
</DocumentFragment>
`;

exports[`Logo to match snapshot ["shifter"] 1`] = `
<DocumentFragment>
  <img
    alt="Shifter"
    height="66.66666666666666"
    width="50"
  />
</DocumentFragment>
`;
